auth/default_authorizer.hh

```diff
@@ -25,10 +26,11 @@ namespace auth {
 
 class default_authorizer : public authorizer {
     cql3::query_processor& _qp;
+    ::service::raft_group0_client& _group0_client;
 
     ::service::migration_manager& _migration_manager;
 
-    abort_source _as{};
+    mutable abort_source _as{};
```

Comment:
because I use it in const functions (e.g. for some reason password_authenticator::create is declared as const) and for some other reason add_entry requires non const abort source.

The const methods are part of some bigger pure virtual interface class, so I chose "lesser evil"

Reply:
Please fix the authenticator interface (best to do this in a separate patch before this series to retire it).

These mutables are time bombs, better not to add them.

Reply:
Ok, will fix in a separate PR. It's a bit unfortunate that using abort source requires method to be non const, as I understand this is a tradeoff to make subscribe() possible lower in the code.

Reply:
Created https://github.com/scylladb/scylladb/pull/17312